README.md

```diff
@@ -1,181 +1,115 @@
-      Picasso.get()
-        .load("https://upload.wikimedia.org/wikipedia/en/9/92/BenSisko.jpg")
-        .into(this)
-    }
+  private val bio = TextView(context).apply {
+    maxLines = 2
+    text = "..."
+  }
 
   init {
     avatar.layoutBy(
-      x = leftTo { name.left() }
-        .widthOf { name.width() },
-      y = topTo { name.bottom() }
-        .heightOf { name.width().toY() }
+      x = leftTo { parent.left() }.widthOf { 60.xdip },
+      y = topTo { parent.top() }.heightOf { 60.ydip }
     )
+    contourHeightOf { bio.bottom() }
   }
+}
 ```
 
-Finally, let's insert a created date between the note content and the bottom of the view. If there is not enough content in the `note: TextView`, let's align the created date vertically with the name & icon.
+#### Runtime Layout logic
 
-```kotlin
-  private val starDate = TextView(context).apply {
-    text = "Stardate: 23634.1"
-    setTextColor(White)
-    setTextSize(TypedValue.COMPLEX_UNIT_DIP, 18f)
-  }
+Because you're writing plain Kotlin, you can reference other Views and use maths in your lambdas for describing your layout. These lambdas can also be used for making runtime decisions that will be evaluated on every layout pass.
 
-  init {
-    starDate.layoutBy(
-      x = rightTo { parent.right() - 15.dip },
-      y = maxOf(
-        topTo { note.bottom() + 5.dip },
-        bottomTo { name.bottom() }
-      )
-    )
+```kotlin
+bio.layoutBy(
+  x = ...
+  y = topTo {
+    if (isSelected) avatar.top() + 16.ydip
+    else avatar.centerY() - bio.preferredHeight() / 2
+  }.heightOf {
+    if (isSelected) bio.preferredHeight()
+    else 100.ydip
   }
+)
 ```
 
-What does the end result of this look like?
-
-<p align="center">
-  <img width="460" src="screenshots/simple_demo.gif">
-</p>
+When paired with an animator, your runtime layout decisions can even act as keyframes for smoothly updating your layout.
 
-## Features
-### Functional API
-Similar to `ConstraintLayout`, Contour works by creating relationships between views' position and size. The difference is, instead of providing opaque XML attributes, you provide functions which are directly called during the layout phase.
-This offers a couple advantages:
-
-#### Runtime Layout Logic
-Since configuration is simply provided through lambdas, you can make runtime layout decisions.
-For example:
-```kotlin
-leftTo {
-  if (user.name.isEmpty) parent.left()
-  else nameView.right()
-}
-```
-Or
 ```kotlin
-heightOf {
-  maxOf(note.height(), 100.dip)
+setOnClickListener {
+  TransitionManager.beginDelayedTransition(this)
+  isSelected = !isSelected
+  requestLayout()
 }
 ```
 
-#### Operators
-You can use any of the Kotlin operators in your layouts for defining spacing, padding, etc.
-```kotlin
-rightTo { parent.right() - 15.dip }
-```
-Or
-```kotlin
-centerHorizontallyTo { parent.width() / 4 }
-```
+What does the end result of this look like?
+
+![runtime layout decision](screenshots/runtime_layout_decisions.gif)
```

Comment:
You might want to disable `clipToPadding` in the GIF. The bottom padding clipping the second line of text distracts me from what's actually happening.

Reply:
There's also a space missing between "actor" "and"

Reply:
> The bottom padding clipping the second line of text distracts me from what's actually happening.

True it looks ugly but disabling clipping doesn't fix it. I'll try spending more time on this.

Reply:
Not a blocker.

Reply:
I managed to make it slightly better: https://github.com/cashapp/contour/raw/saket/sep10/readme-and-sample-updates/screenshots/runtime_layout_logic.gif

Reply:
Huh. Something is clipping it. Maybe I can look today. I have a bunch of meetings. It's just the sample in this PR right?

Reply:
I suspect it's the transitions framework doing something, but I didn't spend enough time investigating it. Let me push a branch for you.

Reply:
Here you go: `saket/sep15/layout-clipping`

Reply:
Seems like a byproduct of ChangeBounds which is used by AutoTransition. The height is reduced beyond what can display two lines and it crops for some reason (despite setting clipToPadding=false).

Reply:
Yeah that's a shame. I'll go ahead and merge this and maybe we can try making it better in the future?

Reply:
Yep!